lib/mixins/property-effects.html

```diff
@@ -1292,32 +1288,28 @@
        */
       _setPendingPropertyOrPath(path, value, shouldNotify, isPathNotification) {
         let rootProperty = Polymer.Path.root(Array.isArray(path) ? path[0] : path);
-        let hasAccessor = this.__dataHasAccessor && this.__dataHasAccessor[rootProperty];
-        let isPath = (rootProperty !== path);
-        if (hasAccessor) {
-          if (isPath) {
-            if (!isPathNotification) {
-              // Dirty check changes being set to a path against the actual object,
-              // since this is the entry point for paths into the system; from here
-              // the only dirty checks are against the `__dataTemp` cache to prevent
-              // duplicate work in the same turn only. Note, if this was a notification
-              // of a change already set to a path (isPathNotification: true),
-              // we always let the change through and skip the `set` since it was
-              // already dirty checked at the point of entry and the underlying
-              // object has already been updated
-              let old = Polymer.Path.get(this, path);
-              path = /** @type {string} */ (Polymer.Path.set(this, path, value));
-              // Use property-accessor's simpler dirty check
-              if (!path || !super._shouldPropertyChange(path, value, old)) {
-                return false;
-              }
+        if (rootProperty !== path) {
+          if (!isPathNotification) {
+            // Dirty check changes being set to a path against the actual object,
+            // since this is the entry point for paths into the system; from here
+            // the only dirty checks are against the `__dataTemp` cache to prevent
+            // duplicate work in the same turn only. Note, if this was a notification
+            // of a change already set to a path (isPathNotification: true),
+            // we always let the change through and skip the `set` since it was
+            // already dirty checked at the point of entry and the underlying
+            // object has already been updated
+            let old = Polymer.Path.get(this, path);
+            path = /** @type {string} */ (Polymer.Path.set(this, path, value));
+            // Use property-accessor's simpler dirty check
+            if (!path || !super._shouldPropertyChange(path, value, old)) {
+              return false;
             }
-            this.__dataHasPaths = true;
           }
+          this.__dataHasPaths = true;
```

Comment:
For paths, the only time `Polymer.Path.set` is required to update the value is _iff_ it is `!isPathNotification`; if it is a path notification, the value is guaranteed to be set.